dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultIconService.java

```diff
@@ -72,8 +72,9 @@ public class DefaultIconService implements IconService {
 
   @Override
   @Transactional(readOnly = true)
-  public List<Icon> getIcons() {
-    return Stream.concat(defaultIcons.values().stream(), customIconStore.getAllIcons().stream())
+  public List<Icon> getIcons(IconCriteria iconCriteria) {
```

Comment:
should we add a couple of tests to validate the pagination works?

Reply:
on the way